contrib/mesos/pkg/scheduler/service/service.go

```diff
@@ -559,9 +568,8 @@ func (s *SchedulerServer) Run(hks hyperkube.Interface, _ []string) error {
 		validation := ha.ValidationFunc(validateLeadershipTransition)
 		srv := ha.NewCandidate(schedulerProcess, driverFactory, validation)
 		path := fmt.Sprintf(meta.DefaultElectionFormat, s.frameworkName)
-		sid := uid.New(eid.Group(), "").String()
-		log.Infof("registering for election at %v with id %v", path, sid)
-		go election.Notify(election.NewEtcdMasterElector(etcdClient), path, sid, srv, nil)
+		log.Infof("registering for election at %v with id %v", path, eid.GetValue())
+		go election.Notify(election.NewEtcdMasterElector(etcdClient), path, eid.GetValue(), srv, nil)
```

Comment:
bug #787: `sid` used to contain a UUID so that each scheduler registered a unique name. this PR breaks multi-scheduler HA because identical executor ID's `eid` are reused as the scheduler UUID across multiple schedulers.